Scrubber Re-balance

MrsTonedOne opened the pull request:
Scrubber buffs (and nerfs)

Scrubbers now use power.
How much depends on what it's doing, how many gases its set to filter, etc.
Scrubbers can now be configured (via the air alarm) to pull from a 3x3 area around them at a major power cost!

Air alarms given some new environmental modes to account for this.

Contaminated - Scrubs out ALL contaminants quickly (all filters on + 3x3)
Refill - Triple vent output

Panic siphon now uses 3x3 range rather than changing the scrubber's volume rate to 2000.

Added in siphon mode, it's panic siphon without the panic. (no 3x3)

Fixed some bugs relating to atmos and plasma spreading:

Linda will be a lot less aggressive about sleeping active turfs it thinks aren't active enough. This should prevent stray pockets of plasma not going away until you drag a portable scrubber under them.

Turfs are more aggressive about detecting what their overlay should be, this will prevent stray pockets of plasma overlays that aren't really there (or vise versa)

Edit: 3x3 has been added to the huge portable scrubber, they will always pull from a 3x3 range instead of their volume rate buff

MrsTonedOne committed:
Scrubber tweaks + related air alarm tweaks.
Scrubbers now use power.
How much depends on what its doing, how many gases its set to filter, etc.
Scrubbers can now pull from a 3x3 area around them at a major power cost!

Air alarms given some new environmental modes to account for this.

MrsTonedOne committed:
Fixes bug with atmos overlays not updating.

MrsTonedOne committed:
Tweaks some atmos defines relating to spreading and plasma
Makes plasma show up as an overlay more easily, and makes linda less aggressive about sleeping active turfs.

Jalleo commented:
:+1:  Title could be "Scrubber Re-balance" though

MrStonedOne commented:
20w a scrubber at round start.

but if you turn all filtering on, that jumps to 40.

Siphoning? that jumps to 60.

3x3 filtering on the other hand uses more at a rate that is expodental to the number of atmos accessible turfs in the range around around it.

So 3x3 filtering + siphon on a full range of tiles hits lower 4digits.

GunHog commented:
Finally happy to see them being useful! :+1:

RemieRichards commented:
Excellent :+1:

Cebasiz commented:
i always love atmos changes

seems good, the power drain shouldn't be too bad; SMES were changed a while back to have an enormous power reserve at roundstart and stage3 singulo should still power 5 stations on it's supply so whatever.

duncathan commented:
:+1:
I don't even care that it's gonna conflict with my PR; dealing with merge conflicts is worth it to get this

duncathan commented:
Does this fix #9787?

RemieRichards commented:
@duncathan I forget the specifics of the issue but, whether it's just a visual overlay of the gas or genuine deadlocking:

>Linda will be a lot less aggressive about sleeping active turfs it thinks aren't active enough. This should prevent stray pockets of plasma not going away until you drag a portable scrubber under them.

>Turfs are more aggressive about detecting what their overlay should be, this will prevent stray pockets of plasma overlays that aren't really there (or vise versa)

(form the PR body)

Aranclanos commented on code/LINDA/LINDA_turf_tile.dm:
```diff
@@ -6,6 +6,7 @@
 	var/atmos_adjacent_turfs = 0
 	var/atmos_adjacent_turfs_amount = 0
 	var/atmos_supeconductivity = 0
+	var/atoms_overlay_type = ""
```
don't recreate this variable, use the one in gas mixtures datum.
Also, if its better, remove the one from gas mixtures and let this one exist, no other atom uses it if I'm not wrong, but doublecheck that because I'm not sure.
About performance, I don't remember how the gas mixtures edit the var, so thats why im saying.

freerealestate commented:
:+1: Hopefully this will make mass chemical spills/breaches more feasible to fix, instead of causing a shuttle call because no one knows how to fix them

Alek2ander commented:
Can the big portable scrubbers (the ones in toxins storage) always scrub 3x3?

TheVekter commented:
>Linda will be a lot less aggressive about sleeping active turfs it thinks aren't active enough. This should prevent stray pockets of plasma not going away until you drag a portable scrubber under them.

Holy shit that's an old bug, thanks for finally fixing it.

GunHog commented:
How does "REPLACE AIR" work now? Is there a 3x3 scrub + refill option in the air alarm? This mode would work very well in cleaning up plasma fires that started from canisters.

MrsTonedOne committed:
Removes gas mixture graphic overlay code, Moves all of that to turfs.
(also fixes conflicts)

Conflicts:
	code/LINDA/LINDA_turf_tile.dm

MrStonedOne commented:
>How does "REPLACE AIR" work now?

The code considers it the same as a panic siphon for the first part, so its uses 3x3.

But it does not use refill air, because then I would need to have it detect when its full and revert to normal, and i figured doing that would be automating too much.

MrStonedOne commented:
>Can the big portable scrubbers (the ones in toxins storage) always scrub 3x3?

Hmm, I could do that, yes.

MrsTonedOne committed:
Fixes turfs updating their atmos overlay every atmos tick
(I forgot a line)
Also makes turfs only remove the current atmos overlay, not all of them.
(I had tried this before, but it didn't work, and that turns out to be because of the line I forgot)

MrsTonedOne committed:
Gives the large portable scrubber 3x3 scurbbing power
Also fixes portable scrubbers a bit

Conflicts:
	code/ATMOSPHERICS/components/unary_devices/vent_scrubber.dm

MrsTonedOne committed:
Fixed scrubbers runtiming

MrsTonedOne committed:
change log
